src/core/net/dns_headers.hpp

```diff
@@ -382,6 +619,54 @@ OT_TOOL_PACKED_BEGIN
 class ResourceRecord
 {
 public:
+    /**
+     * Resource Record Types.
+     *
+     */
+    enum : uint16_t
+    {
+        kTypeA    = 1,  ///< Address record (IPv4).
+        kTypePtr  = 12, ///< PTR record.
+        kTypeTxt  = 16, ///< TXT record.
+        kTypeSrv  = 33, ///< Service locator record.
+        kTypeAaaa = 28, ///< IPv6 address record.
+        kTypeKey  = 25, ///< Key record.
+        kTypeOpt  = 41, ///< Option record.
+    };
+
+    /**
+     * Resource Record Class Codes.
+     *
+     */
+    enum : uint16_t
+    {
+        kClassInternet = 1, ///< Class code Internet (IN).
+    };
+
+    /**
+     * This method initializes the resource record.
+     *
+     * @param[in] aType   The type of the resource record.
+     * @param[in] aClass  The class of the resource record (default is `kClassInternet`).
+     * @param[in] aTtl    The time to live field of the resource record (default is zero).
+     *
+     */
+    void Init(uint16_t aType, uint16_t aClass = kClassInternet, uint32_t aTtl = 0);
```

Comment:
Do we have a use case for other class values?

Reply:
There is use cases in [DNS UPDATE](https://tools.ietf.org/html/rfc2136) that the class is used to decide the type of a DNS update (the `class` is compared against `ANY CLASS` and `NONE CLASS`).

Reply:
I suggest we keep the method as is for now. We can optimize it later.